images/Dockerfile.agent

```diff
@@ -21,4 +21,13 @@ WORKDIR /
 COPY --from=builder /workspace/bin/infraagent /infraagent
 COPY --from=builder /workspace/bin/felix-api-proxy /felix-api-proxy
 
+COPY certs/infra-server-ca.crt /etc/pki/tls/infra/
+COPY certs/infra-server-ca.key /etc/pki/tls/infra/
+COPY certs/infra-client-ca.crt /etc/pki/tls/infra/
+COPY certs/infra-client-ca.key /etc/pki/tls/infra/
+COPY certs/infraagent-server.crt /etc/pki/tls/infra/
+COPY certs/infraagent-server.key /etc/pki/tls/infra/
+COPY certs/infraagent-client.crt /etc/pki/tls/infra/
+COPY certs/infraagent-client.key /etc/pki/tls/infra/
```

Comment:
I don't think this is a good idea to add tls certs and key in container images. It's equivalent to adding you password file in container image which can be read by anyone to extract them.

Reply:
Yes. The location of the certs is still under discussion. The changes here are draft and only for development.

Reply:
It's not about location. It's about not adding them in the container. The tls key and certs should be provided at runtime via CLI params or config file option for their location. The code should read from those location or look in some default location. But the actual tls files should not be included in the container image. Instead we could use volume mount where those files will be added when pods are run.

Reply:
Yes. I meant the same.

Reply:
Agree ---- the certificates should not be part of deployment image if the solution needs to remain in-line with SDLe principles.

Reply:
We have a config file option in Manager. Is agent config file also working?

Reply:
Yes. The agent have infraagent.yaml file. We pass interfaceType(tap,sriov etc) and inframanagerAuthType through the file.

Reply:
For the inframanager, I have added an option to the config.yaml file to choose the authentication type. It can be one of insecure, server side TLS or mutual TLS.